wsmaster/che-core-api-devfile/src/main/resources/schema/devfile.json

```diff
@@ -1,49 +1,59 @@
 {
+  "meta:license" : [
+    " Copyright (c) 2012-2019 Red Hat, Inc.",
+    " This program and the accompanying materials are made",
+    " available under the terms of the Eclipse Public License 2.0",
+    " which is available at https://www.eclipse.org/legal/epl-2.0/",
+    " SPDX-License-Identifier: EPL-2.0",
+    " Contributors:",
+    " Red Hat, Inc. - initial API and implementation"
+  ],
+  "$schema": "http://json-schema.org/draft-07/schema#",
+  "type": "object",
+  "title": "Devfile object",
+  "description" : "This schema describes the structure of the devfile object",
   "definitions": {
     "attributes" : {
-      "id": "propertyList",
       "type": "object",
       "additionalProperties": {
         "type": "string"
       },
       "javaType": "java.util.Map<String, String>"
     },
     "selector" : {
-      "id": "selector",
       "type": "object",
       "additionalProperties": {
         "type": "string"
       },
       "javaType": "java.util.Map<String, String>"
     }
   },
-  "$schema": "http://json-schema.org/draft-07/schema#",
-  "type": "object",
-  "title": "The Root Schema of DevFile object",
   "required": [
     "specVersion",
     "name",
     "tools"
   ],
+  "additionalProperties": false,
   "properties": {
     "specVersion": {
       "type": "string",
-      "title": "Devfile Schema Specification Version",
+      "title": "Devfile Specification Version",
       "examples": [
         "0.0.1"
       ]
     },
     "name": {
       "type": "string",
-      "title": "Devfile Name",
+      "title": "Devfile Name.",
+      "description" : "The name of the devfile. Workspaces created from devfile, will inherit this name",
```

Comment:
Maybe it makes sense to add a few words about using generate name if a workspace with such name already exists.

Reply:
This is too Che specific IMO. Not related to the schema itself...